relay-server/src/utils/rate_limits.rs

```diff
@@ -273,12 +275,11 @@ impl Enforcement {
         self.event.is_active()
     }
 
-    /// Helper for `track_outcomes`.
-    fn get_outcomes(
-        self,
-        envelope: &Envelope,
-        scoping: &Scoping,
-    ) -> impl Iterator<Item = TrackOutcome> {
+    /// Invokes [`TrackOutcome`] on event-related enforcements reported by the [`EnvelopeLimiter`].
+    ///
+    /// Datatypes with 1:1 relationships between outcomes and enforcement are handled automatically
+    /// by retain_items.
+    fn track_event_outcomes(&self, envelope: &Envelope, scoping: &Scoping) {
```

Comment:
Eventually I would like to get rid of this function and emit all outcomes from `retain_item`.